boa/src/syntax/parser/mod.rs

```diff
@@ -202,8 +244,9 @@ impl Parser {
         }
     }
 
-    /// Peek the next token and if it is ``kind``, get the next token, return true.
-    /// Otherwise, return false.
+    /// Peek the next token, if it's of `kind` type.
+    ///
+    /// When the next token is a `kind` token, get the token, otherwise return `None`.
     fn next_if(&mut self, kind: TokenKind) -> Option<Token> {
         match self.peek(0) {
             Ok(tok) => {
```

Comment:
The bug is here. It parses it correctly `var x = [1, 2, 3];` now.
```suggestion
    /// Gets the next token, if the current token is of `kind` type.
    ///
    /// When the current token is a `kind` token, get the next token, otherwise return `None`.
    fn next_if(&mut self, kind: TokenKind) -> Option<Token> {
        match self.get_current_token() {
            Ok(tok) => {
```

Reply:
The previous implementation was more like `if_next` rather than `next_if`!

Any thoughts on the matter? :thinking:

Reply:
Interesting,
So next_if should only consume the next token if it matches, if it doesn’t then it shouldn’t move forward. That was the idea behind it, I guess getting the cure token is fine, but if it matches you should move forward 1.
(On my phone atm) doesn’t peek(0) and get current token effectively do the same thing?

Reply:
The implementation of peek is:
```rust
/// peeks at the next token
fn peek(&self, num: usize) -> Result<Token, ParseError> {
   self.get_token(self.pos + 1 + num)
}
```
so even if it is `peek(0)` it gives the next token, not the current.

Reply:
What you have is more correct now that next gets the current token then moves forward

Reply:
So yes you’re right now that next gas changed we need to change this too

Reply:
I think we should change the methods `next_if` to `matches` and `next_if_skip_lineterminator` to `matches_skip_lineterminator`. Which better aligns with the implementation of the functions.

Reply:
Matches is a little bit ambiguous to me, it’s not clear that it would advance the parser if there’s a match, it sounds like a function that just checks if there’s a match between 2 things or not.

next_if was clear in the sense it would advance if the value was correct

Reply:
> Matches is a little too ambiguous for me, it’s not clear that it would advance the parser if there’s a match, it sounds like a function that just checks if there’s a match between 2 things or not.
> 
> next_if was clear in the sense it would advance if the value was correct

I see! Thats a good point.